src/transformers/models/oneformer/processing_oneformer.py

```diff
@@ -37,34 +70,43 @@ class OneFormerProcessor(ProcessorMixin):
             The image processor is a required input.
         tokenizer ([`CLIPTokenizer`, `CLIPTokenizerFast`]):
             The tokenizer is a required input.
-        max_seq_len (`int`, *optional*, defaults to 77)):
+        max_seq_length (`int`, *optional*, defaults to 77):
             Sequence length for input text list.
-        task_seq_len (`int`, *optional*, defaults to 77):
+        task_seq_length (`int`, *optional*, defaults to 77):
             Sequence length for input task token.
     """
 
     attributes = ["image_processor", "tokenizer"]
     image_processor_class = "OneFormerImageProcessor"
     tokenizer_class = ("CLIPTokenizer", "CLIPTokenizerFast")
+    optional_call_args = ["task_inputs", "segmentation_maps"]
 
     def __init__(
-        self, image_processor=None, tokenizer=None, max_seq_length: int = 77, task_seq_length: int = 77, **kwargs
+        self,
+        image_processor=None,
+        tokenizer=None,
     ):
         if image_processor is None:
             raise ValueError("You need to specify an `image_processor`.")
         if tokenizer is None:
             raise ValueError("You need to specify a `tokenizer`.")
 
-        self.max_seq_length = max_seq_length
-        self.task_seq_length = task_seq_length
-
         super().__init__(image_processor, tokenizer)
 
-    def _preprocess_text(self, text_list=None, max_length=77):
-        if text_list is None:
-            raise ValueError("tokens cannot be None.")
-
-        tokens = self.tokenizer(text_list, padding="max_length", max_length=max_length, truncation=True)
+    def _preprocess_text(
+        self,
+        text_list: PreTokenizedInput,
+        max_length: Optional[int] = None,
+        text_kwargs: Optional[dict] = None,
+    ):
+        if text_kwargs is None:
+            text_kwargs = {}
+        tokens = self.tokenizer(
+            text_list,
+            max_length=max_length if max_length is not None else text_kwargs.get("max_length", 77),
```

Comment:
I would say the opposite:
```suggestion
            max_length=text_kwargs.get("max_length") if text_kwargs.get("max_length") is not None else max_length,
```
So that the `max_length` kwarg overrides the default kwarg when it is specified